jlm/rvsdg/region.hpp

```diff
@@ -380,6 +380,13 @@ class Region
   void
   append_argument(RegionArgument * argument);
 
+  /* \brief Insert \p argument into argument list of the region
+   *
+   * Multiple invocations of append_argument for the same argument are undefined.
+   */
+  void
+  insert_argument(size_t index, RegionArgument * argument);
+
```

Comment:
Why is that all in a sudden required?

Reply:
To maintain relative ordering of context vars / loop vars / fixpoint vars, when building objects incrementally. Agree that this is not the most desirable way to do it, but it at least allows to transition the APIs for the moment.